admin/admin-block-editor.php

```diff
@@ -48,11 +48,7 @@ public function filter_preload_paths( $preload_paths, $context ) {
 			return $preload_paths;
 		}
 
-		// Backward compatibility with WP < 6.0 where `WP_Block_Editor_Context::$name` doesn't exist yet.
-		if (
-			( property_exists( $context, 'name' ) && 'core/edit-post' !== $context->name )
-			|| ! $context->post instanceof WP_Post
-		) {
+		if ( 'core/edit-post' !== $context->name ) {
```

Comment:
```suggestion
		if ( 'core/edit-post' !== $context->name || ! $context->post instanceof WP_Post ) {
```
So it behaves like before.

Reply:
Only `property_exists()` was here for backward compatibility.
Ensuring `$context->post` is a `WP_Post` is type safety.
On the other hand, `$context->name` should be `'core/edit-post'` regarding the screen to filter.
Because it could be badly set or change on the fly since it's a public prop.
See the doc [here](https://github.com/WordPress/wordpress-develop/blob/31ce381cc8a0f63af53c5f107f626763c3d29cb1/src/wp-includes/class-wp-block-editor-context.php#L39).

Reply:
Bonus, it will remove PHPStan errors 😅